core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1260,16 +1261,19 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
             }
 
             QualifiedObjectName name = createQualifiedObjectName(session, table, table.getName());
-            analysis.addEmptyColumnReferencesForTable(accessControl, session.getIdentity(), name);
             Optional<TableHandle> tableHandle = Optional.empty();
+            QualifiedObjectName targetTableName = name;
 
             Optional<ConnectorMaterializedViewDefinition> optionalMaterializedView = metadata.getMaterializedView(session, name);
             if (optionalMaterializedView.isPresent()) {
+                analysis.addEmptyColumnReferencesForTable(accessControl, session.getIdentity(), name);
                 if (metadata.getMaterializedViewFreshness(session, name).isMaterializedViewFresh()) {
                     // If materialized view is current, answer the query using the storage table
                     Optional<QualifiedName> storageName = getMaterializedViewStorageTableName(optionalMaterializedView.get(), name);
                     if (storageName.isPresent()) {
-                        tableHandle = metadata.getTableHandle(session, createQualifiedObjectName(session, table, storageName.get()));
+                        QualifiedObjectName storageTableName = createQualifiedObjectName(session, table, storageName.get());
+                        checkStorageTableNotRedirected(storageTableName);
```

Comment:
why?

Reply:
I think we can force connector to not redirect the storage table, since it can be writable and we've avoided similar usecases for now.